go/tasks/pluginmachinery/k8s/config.go

```diff
@@ -0,0 +1,58 @@
+package k8s
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	restclient "k8s.io/client-go/rest"
+)
+
+type ClusterConfig struct {
+	Name     string `json:"name" pflag:",Friendly name of the remote cluster"`
+	Endpoint string `json:"endpoint" pflag:", Remote K8s cluster endpoint"`
+	Auth     Auth   `json:"auth" pflag:"-, Auth setting for the cluster"`
+	Enabled  bool   `json:"enabled" pflag:", Boolean flag to enable or disable"`
+}
+
+type Auth struct {
+	Type      string `json:"type" pflag:", Authentication type"`
```

Comment:
What are the possible values? or maybe a link to the docs that list the values?
```suggestion
	Type      string `json:"type" pflag:",Authentication type"`
```

Reply:
aha, make sense... let's remove it then as you said... addition is backward compat, removing/changing type... etc. isn't...